src/scaffold/entry_type/integrity.rs

```diff
@@ -531,375 +463,311 @@ fn add_entry_type_to_validation_arms(
+                            }}
+                        }},"#,
+                    ))?;
+                    entry_type_match.arms.push(new_arm);
+                }
+            } else if is_entry(&op_record_arm.pat, "DeleteEntry") {
+                // Add new entry type to match arm
+                if find_ending_match_expr(&mut op_record_arm.body).is_none() {
+                    // Change empty invalid to match on entry_type
+                    *op_record_arm.body = syn::parse_quote! {
+                        {
+                            let original_record = must_get_valid_record(original_action_hash)?;
+                            let original_action = original_record.action().clone();
+                            let original_action = match original_action {
+                                Action::Create(create) => EntryCreationAction::Create(create),
+                                Action::Update(update) => EntryCreationAction::Update(update),
+                                _ => {
+                                    return Ok(ValidateCallbackResult::Invalid(
+                                        "Original action for a delete must be a Create or Update action".to_string()
+                                    ));
+                                }
+                            };
+                            let app_entry_type = match original_action.entry_type() {
+                                EntryType::App(app_entry_type) => app_entry_type,
+                                _ => {
+                                    return Ok(ValidateCallbackResult::Valid);
+                                }
+                            };
+                            let entry = match original_record.entry().as_option() {
+                                Some(entry) => entry,
+                                None => {
+                                    return Ok(ValidateCallbackResult::Invalid(
+                                        "Original record for a delete must contain an entry".to_string(),
+                                    ));
+                                }
+                            };
+                            let original_app_entry = match EntryTypes::deserialize_from_type(
+                                app_entry_type.zome_index,
+                                app_entry_type.entry_index,
+                                entry,
+                            )? {
+                                Some(app_entry) => app_entry,
+                                None => {
+                                    return Ok(ValidateCallbackResult::Invalid(
+                                        "Original app entry must be one of the defined entry types for this zome"
+                                            .to_string(),
+                                    ));
+                                }
+                            };
+                            match original_app_entry {}
```

Comment:
This looks like it's not of any use anymore, is it?

Reply:
It's still in use, the code right below it populates the match arm

Reply:
Ah I see, this makes sense.